tools/indexer/wallet/src/main.rs

```diff
@@ -0,0 +1,221 @@
+                _ => None,
+            })
+            .collect::<Vec<&near_indexer::near_primitives::views::ExecutionOutcomeWithIdView>>();
+
+        diesel::update(
+            schema::access_keys::table.filter(
+                schema::access_keys::dsl::receipt_hash.eq_any(
+                    receipt_outcomes
+                        .iter()
+                        .filter(|outcome| match &outcome.outcome.status {
+                            near_indexer::near_primitives::views::ExecutionStatusView::Failure(
+                                _,
+                            ) => true,
+                            _ => false,
+                        })
+                        .map(|outcome| outcome.id.clone().to_string())
+                        .collect::<Vec<String>>(),
+                ),
+            ),
+        )
+        .set(schema::access_keys::dsl::status.eq(FAILED))
+        .execute_async(&pool)
+        .await
+        .unwrap();
+
+        diesel::update(
+            schema::access_keys::table.filter(
+                schema::access_keys::dsl::receipt_hash.eq_any(
+                    receipt_outcomes
+                        .iter()
+                        .filter(|outcome| match &outcome.outcome.status {
+                            near_indexer::near_primitives::views::ExecutionStatusView::SuccessReceiptId(
+                                _,
+                            ) | near_indexer::near_primitives::views::ExecutionStatusView::SuccessValue(_) => true,
+                            _ => false,
+                        })
+                        .map(|outcome| outcome.id.clone().to_string())
+                        .collect::<Vec<String>>(),
+                ),
+            ),
+        )
+        .set(schema::access_keys::dsl::status.eq(SUCCESS))
+        .execute_async(&pool)
+        .await
+        .unwrap();
+
+        // Handle receipts
+        for chunk in &block.chunks {
+            diesel::insert_into(schema::access_keys::table)
+                .values(
+                    chunk
+                        .receipts
+                        .iter()
+                        .filter(|receipt| match receipt.receipt {
+                            near_indexer::near_primitives::views::ReceiptEnumView::Action {
+                                ..
+                            } => true,
+                            _ => false,
+                        })
+                        .map(|receipt| {
+                            AccessKey::from_receipt_view(receipt, block.block.header.height.clone())
+                        })
+                        .flatten()
+                        .collect::<Vec<AccessKey>>(),
+                )
+                .execute_async(&pool)
+                .await
+                .unwrap();
+        }
+    }
+}
+
+fn main() {
+    let home_dir: Option<String> = env::args().nth(1);
+
+    let indexer = near_indexer::Indexer::new(home_dir.as_ref().map(AsRef::as_ref));
```

Comment:
I realized that we want to be able to optionally specify the block height where we want to start. Currently, we start at the first block we observe as final after we synced the network... We should have at least three options (enum): Latest Final (this is the current behavior), Genesis (makes sense for most of the Indexers we are going to implement), BlockHeight / BlockHash (Indexers should be able to fetch the information from their internal DB and pass it to Indexer before it starts)

Reply:
AFAIU it should be done on NEAR Indexer side first of all, how about separate issue for this?